### DISCUSSION TITLE
can't return promise as state

### DISCUSSION THREAD

**Discussion by baixiaoyu2997** [vuejs/pinia#1184] [Help and Questions]
### Reproduction

https://codesandbox.io/s/deorderbook-njtv83?file=/composables/useWallet.js

### Steps to reproduce the bug

look Reproduction

### Expected behavior

store return promise state

### Actual behavior

return blank object

**Comment by posva** [comment]
since promises are not serializable, they need special treatment if you need them on server and client side like in your example. You can use https://pinia.vuejs.org/api/interfaces/pinia.DefineStoreOptions.html#hydrate to customize how the state is hydrated

**Reply by baixiaoyu2997** [reply]
thanks , hydrate is only for DefineStoreOptions?  is there a solution  for `DefineSetupStoreOptions`?

**Reply by posva** [reply]
It’s in the third argument inside an object
